src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -459,8 +459,43 @@ def verify_property(instance, condition):
 index_or_filter_regex = re.compile(r'\[(.*)\]')
 
 
+def _split_key_value_pair(expression):
+
+    def _find_split():
+        """ Find the first = sign to split on (that isn't in [brackets])"""
+        key = []
+        value = []
+        brackets = False
+        chars = list(expression)
+        while len(chars) > 0:
+            c = chars.pop(0)
+            if c == '=' and not brackets:
+                # keys done the rest is value
+                value = chars
+                break
+            elif c == '[':
+                brackets = True
+                key += c
+            elif c == ']' and brackets:
+                brackets = False
```

Comment:
Nesting cannot occur. I would need the indices of all occurrences of [, =, and ] and then find an = that isn't in between [ ]. I figure getting all occurrences of those characters would be 3n and then processing all the logic to figure out if I had a valid = sign would just be more confusing and less efficient than going through the whole string character by character, which is worst case n.